devito/passes/iet/parpragma.py

```diff
@@ -486,6 +495,27 @@ def _make_sections_from_imask(cls, f, imask):
             sections.append('[%s:%s]' % (start, size))
         return ''.join(sections)
 
+    @classmethod
+    def _make_offset_from_imask(cls, f, imask):
+        datasize = cls._map_data(f)
+        if imask is None:
+            imask = [FULL]*len(datasize)
+        assert len(imask) == len(datasize)
+        offset = 0
+        shape = []
+        for n, (i, j) in enumerate(zip(imask, datasize)):
+            if i is FULL:
+                shape.append(j)
+            else:
+                nofs, size = i
+                if isinstance(nofs, Mod):
+                    # NOTE: work around alleged SymPy bug which doesn't allow
```

Comment:
As said on slack, it's not a sympy bug, it's a choice of devito to go with `%` while sympy uses `fmod` which is a function call and therefore always has precedence over Mul.
I think it makes much more sense to fix `print_Mod` rather than adding this odd extra type.

Reply:
It's actually very easy to fix if you use our`Mod` instead of the sympy one and add
```
@property
def precedence(self):
    return 49  # precedence(Mul)-1
```

since the first thing sympy checks is if it's defined before going to the defaults
https://github.com/sympy/sympy/blob/126f80578140e752ad5135aac77b8ff887eede3e/sympy/printing/precedence.py#L123

Reply:
right. However, as explained on Slack, I forgot that this piece of code is actually dead, so I just dropped it